File: test/stack/min-stack.test.ts
import { MinStack } from '../../src/stack/min-stack';

describe('min stack', () => {
  test('min stack - test case 1', async () => {
    const stack = new MinStack();
    stack.push(-2);
    stack.push(0);
    stack.push(-3);

    expect(stack.getMin()).toBe(-3);

    stack.pop();

    expect(stack.top()).toBe(0);
    expect(stack.getMin()).toBe(-2);
  });
});

File: test/leetcode/stack/min_stack_test.py
from leetcode.stack.min_stack import MinStack


def test_case_1():
    stack = MinStack()
    stack.push(0)
    stack.push(-3)

    assert stack.getMin() == -3

    stack.pop()

    assert stack.top() == 0
    assert stack.getMin() == 0
